SonarQube.TeamBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -169,11 +169,42 @@ private static string GetRulesetFilePath(TeamBuildSettings settings)
 
         private IEnumerable<string> UnpackAdditionalFiles(RoslynExportProfile profile)
         {
-            // TODO
+            Debug.Assert(profile.Configuration != null, "Supplied configuration should not be null");
 
-            return null;
+            List<string> additionalFiles = new List<string>();
+            foreach(AdditionalFile item in profile.Configuration.AdditionalFiles)
+            {
+                string filePath = ProcessAdditionalFile(item);
+                if (filePath != null)
+                {
+                    Debug.Assert(File.Exists(filePath), "Expecting the additional file to exist: {0}", filePath);
+                    additionalFiles.Add(filePath);
+                }
+            }
+
+            return additionalFiles;
         }
 
+        private string ProcessAdditionalFile(AdditionalFile file)
+        {
+            if (string.IsNullOrWhiteSpace(file.FileName))
+            {
+                this.logger.LogDebug(Resources.SLAP_AdditionalFileNameMustBeSpecified);
```

Comment:
The "error cases" here aren't really actionable by the end user so they are debug messages rather than warnings FileName attribute is empty, file already exists.